Software 0.31.3 release notes (#1927)
astronomer/docs: software/release-notes.md

0.31 is the latest stable version of Astronomer Software, while 0.30 remains the latest long-term support (LTS) release. To upgrade to 0.31, see [Upgrade Astronomer](upgrade-astronomer.md). For more information about Software release channels, see [Release and lifecycle policies](release-lifecycle-policy.md). To read release notes specifically for the Astro CLI, see [Astro CLI release notes](https://docs.astronomer.io/astro/cli/release-notes).

## 0.31.3

Release date: February 24, 2023

### Additional improvements

- You can now configure `extraVolumes` and `extraVolumeMounts` in the Alertmanager Helm chart, which can be useful for storing secret credentials for services that read your alerts.

> - You can now configure custom ingress annotations for Houston using `astronomer.houston.ingress.annotation` in the Astronomer Helm chart.

Reviewer: ```suggestion
- You can now use `astronomer.houston.ingress.annotation` in the Astronomer Helm chart to configure custom ingress annotations for Houston.
```

> - You can now upgrade the Airflow Helm chart for individual Deployments by SSHing into Houston and running `yarn upgrade-deployments <deployment-id>`.

Reviewer 2: It's not ssh, it's exec, as in `kubectl exec`. It might be better to avoid the weird term "execing" with something like:

```suggestion
- You can now upgrade the Airflow Helm chart for individual Deployments by running `yarn upgrade-deployments <deployment-id>` from within the houston pod.
```

### Bug fixes 

- Fixed an issue where you could not set `AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER` in a Deployment if you were using an Astronomer Certified image.

> - Fixed an issue where removing a user imported from an IdP group when SCIM is enabled can result in orphaned Deployments and Workspaces.

Reviewer: ```suggestion
- Fixed an issue where removing a user imported from an identity provider (IdP) group with SCIM is enabled can cause orphaned Deployments and Workspaces.
```

> - Fixed an issue where removing a user imported from an identity provider (IdP) group with SCIM enabled caused orphaned Deployments and Workspaces.

Reviewer 3: You can now view orphaned deployments separately whenever users are removed from an identity provider (IdP) group via SCIM

- Fixed a security vulnerability where you could query Elasticsearch logs for a Deployment from a different Deployment.

> - Fixed an issue where authentication tokens were visible in NGinx logs produced by the Software UI.

Reviewer 3: *Nginx

> - Fixed an issue where deploying an image using the `docker/build-push-action` GitHub action could produce errors in Houston that affect the entire Astronomer installation.

Reviewer: ```suggestion
- Fixed an issue where deploying an image with the `docker/build-push-action` GitHub action could produce errors in Houston that affected the entire Astronomer Software installation.
```
